jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -1967,7 +1967,7 @@ public boolean checkAlias(String path, Resource resource)
                 LOG.debug("Aliased resource: {}~={}", resource, resource.getAlias());
 
             // alias checks
-            for (AliasCheck check : getAliasChecks())
```

Comment:
The reason this existed this way was to support folks that wanted to have adjustable (at runtime) list of alias checks.

Reply:
Hmm I thought alias checks could only be added before start but looks like that is not the case.

So maybe we do still need the `CopyOnWriteArrayList`, but we don't need to use `getAliasChecks()` for the unmodifiable list here.

Reply:
I don't have a strong opinion on the solution you are working towards.
Just explaining the history of why things are the way they are in Jetty 11 and earlier.
Up to you to decide if this behavior is worth keeping around or changing for Jetty 12.

To add to this, there once was a desire to have different Alias checks depending on url-pattern.
So that you could reject aliases for say `/archives/*` and reject aliases for `/avatars/*`.

Frankly, only the `ResourceService` should have the alias checks, as that's where all of our serve-static-content lies now (for all EE levels).   The uses of say `ServletContext.getResource(String)` should allow anything that is requested from it as that's programmatic from Servlet APIs, not something that a user request can trigger.   It's up to the webapp developer to decide how to sanitize / evaluate if sane the user request before they call `ServletContext.getResource(String)`.   (Example: Jersey and Spring serve static content themselves, not via the DefaultServlet / ResourceHandler.  But spark uses the ResourceHandler.)

Reply:
I'm not convinced we need to allow getAliasChecks to be overridable, as I don't really see the use-case for a dynamic list, nor is there any context passed on which to make that choice.  So I'm OK with this change..... but equally we could assign a single immutable list as a field and return that.